docs/index.rst

```diff
@@ -16,5 +16,6 @@ Documentation :any:`index <genindex>` and :any:`Module <modindex>` index.
   code_ref/index
   generated/gallery/index
   dev_guide/index
+  dev_guide_release/index
```

Comment:
How does this work? If one is excluded at build time, I thought sphinx would error due to a missing reference.

Reply:
Sphinx just ignores it. I've built it and checked.

Reply:
This isn't what I had in mind, you should be able to remove this once you get the conditional stuff working in the index.rst. That being said if sphinx wont error for this then it will probably make your template logic simpler.

Reply:
Sphinx does raise a warning, which we try to avoid.

Reply:
So, what should be the alternative? Should the whole thing be written using jinga?

Reply:
Good question, it might be worth trying.

Reply:
I think only the index file would need to have jinja templating.

Reply:
yeah, you can template the index file so if it's a dev version the toctree is included and if it isn't then just the link is shown